extensions/interactions/rules.json

```diff
@@ -0,0 +1,186 @@
+{
+  "CodeRepl": {
```

Comment:
What about including it in CodeRepl.py? It could be "rules" member variable. 
From my understanding, it needs to easy access for the backend/frontend.